apps/examples/testcase/ta_tc/security/utc/utc_crypto.c

```diff
@@ -56,31 +58,67 @@ static security_handle g_hnd = NULL;
 
 /**
  * @testcase         utc_crypto_aes_encryption_p
- * @brief            encrypt AES
- * @scenario         encrypt AES
+ * @brief            encrypt AES with IV
+ * @scenario         encrypt AES with IV
  * @apicovered       crypto_aes_encryption
- * @precondition     none
+ * @precondition     AES key should be set in Secure Storage
  * @postcondition    none
  */
 static void utc_crypto_aes_encryption_p(void)
 {
-	int i = 0;
-	unsigned char iv[] = "tamp_iv_value";
-	unsigned int iv_len = strlen((const char*)iv) + 1;
+	unsigned char iv[] = { 0x1a, 0x2b, 0x3c, 0x4d, 0x5e, 0x6f, 0x70, 0x81, 0x92, 0xa3, 0xb4, 0xc5, 0xd6, 0xe7, 0xf8, 0x00 };
+	unsigned int iv_len = 16;
 
-	unsigned char plain_text[] = "plain text";
-	unsigned int plain_text_len = strlen((const char*)plain_text) + 1;
+	unsigned char plain_text[] = { 0x4d, 0x79, 0x20, 0x42, 0x79, 0x74, 0x65, 0x20, 0x50, 0x72, 0x69, 0x6e, 0x74, 0x00, 0x00, 0x00 };
+	unsigned int plain_text_len = 16;
 
 	security_data plain = {plain_text, plain_text_len};
 	security_data enc = {NULL, 0};
 
-	for (; i < sizeof(g_aes_mode_table)/sizeof(security_aes_mode); i++) {
-		security_aes_mode mode = g_aes_mode_table[i];
-		security_aes_param param = {mode, iv, iv_len};
-		security_error res = crypto_aes_encryption(g_hnd, &param, UTC_CRYPTO_KEY_NAME,
-												   &plain, &enc);
-		TC_ASSERT_EQ("crypto_aes_encryption", res, SECURITY_OK);
-		TC_SUCCESS_RESULT();
+	for (int aes_mode = 0; aes_mode < sizeof(g_aes_mode_table) / sizeof(security_aes_mode); aes_mode++) {
```

Comment:
2. What is a difference between g_aes_mode_table and security_aes_mode? Why do you need to define g_aes_mode_table?
3. Could you let me know what you want from sizeof(security_aes_mode)? sizeof(int)? Could you let me know purpose of it?

You didn't reply my previous comments. Please let me get them.

Reply:
I wrote above your review.

2. security_aes_mode is the typedef of aes parameter, and g_aes_mode_table is actual type table to use security_aes_mode.
3. sizeof(g_aes_mode_table) / sizeof(security_aes_mode) is the number of elements of g_aes_mode_table, to test all types of aes mode.

Reply:
2. I can't find any difference between `g_aes_mode_table` and `security_aes_mode`. Why don't you use `security_aes_mode` directly? Why do you need to define `g_aes_mode_table`?
3. Is that a common way for enum? Don't you use a last enum type to get number of member (size)?